ruby/red-arrow/lib/arrow/table.rb

```diff
@@ -327,15 +327,18 @@ def slice(*args)
           to += n_rows if to < 0
           ranges << [from, to]
         when ::Array
-          boolean_array_to_slice_ranges(slicer, 0, ranges)
+          slicer.map! { |value| value.nil? ? false : value }
+          return filter(slicer)
```

Comment:
@kou
What do you think about using `Arrow::Table#filter` in Array, ChunkedArray, and BooleanArray argument cases?
I've converted `nil` to `false` in Array because the behavior of `Arrow::Table#slice` and `Arrow::Table#filter` is different as the following.

#### raw data
```rb
=> #<Arrow::Table:0x7fe2c3236b50 ptr=0x7fe2c25884b0>
        count   visible
0           1   true
1           2   false
2           4
3           8   true
4          16   true
5          32   false
6          64
7         128
```


#### Arrow::Table#slice([nil, true, true, false, true, false, true, true])  -> not including `nil` in the output
```rb
=> #<Arrow::Table:0x7fe2c31c9848 ptr=0x7fe2c27681a0>
        count   visible
0           2   false
1           4
2          16   true
3          64
4         128
```

#### Arrow::Table#filter([nil, true, true, false, true, false, true, true])  -> including `nil` in the output
```rb
=> #<Arrow::Table:0x7fe2c3247680 ptr=0x7fe2c784e040>
        count   visible
0
1           2   false
2           4
3          16   true
4          64
5         128
```

Reply:
Let's use the same behavior as `#filter`.